### DISCUSSION TITLE
Possible to run "All" Microsoft365 Checks?

### DISCUSSION THREAD

**Discussion by n2x4** [silverhack/monkey365#56] [Q&A] [answered]
Hello,

Reviewing the documentation, I see I can run an Azure Instance for "All", but it doesn't look like the same "all" option is supported on the 365 side. I also tried entering all options comma separated: 

Analysis = 'ExchangeOnline,Microsoft365,MicrosoftTeams,Purview,SharePointOnline'; 

But that wasn't supported either. Not sure if this is unsupported and each analysis needs to be run manually. Appreciate any help!

**Comment by silverhack** [comment, marked as answer]
Hi @n2x4, 

This is intentional. Monkey365 requires the use of first-party Microsoft registered applications to connect to Microsoft 365 services when interactive authentication is used, and you may receive multiple authentication prompts. This is different in Azure, in which you can connect to all services within an Azure subscription with one Microsoft's (Microsoft Azure PowerShell ClientId) first-party application. 

Please, also note that the -Analysis is declared as an [Array], and quotation marks are used to specify a **literal string**, instead an Array. The following command should work:

$param = @{
    Instance = 'Microsoft365';
    Analysis = 'ExchangeOnline','Microsoft365';
    PromptBehavior = 'SelectAccount';
    TenantID = '000000-00000000-0000';
}
$assets = Invoke-Monkey365 @param

To avoid misspelling commands, please make use of TAB-completion in PowerShell.

Cheers,
